components/org.wso2.carbon.healthcheck.api.core/src/main/java/org/wso2/carbon/healthcheck/api/core/CarbonMemoryAgentService.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

Comment:
I have a concern having this in healtcheck api core. The reasons is, there are number of carbon dependencies to this core. Hence we won't be able to reuse it in a different java environment. Can we do this as a separate module ?